analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/BazelProfileConstants.java

```diff
@@ -14,15 +14,61 @@
 
 package com.engflow.bazel.invocation.analyzer.bazelprofile;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+
 /** Constants that are specific to Bazel profiles. */
 public class BazelProfileConstants {
   // Thread names: event.name == METADATA_THREAD_NAME && event.args.name == constant below
   public static final String THREAD_CRITICAL_PATH = "Critical Path";
-  public static final String THREAD_GARBAGE_COLLECTOR = "Garbage Collector";
-  public static final String THREAD_MAIN = "Main Thread";
+
+  @VisibleForTesting public static final String THREAD_GARBAGE_COLLECTOR = "Garbage Collector";
+  // See https://github.com/bazelbuild/bazel/commit/a03674e6297ed5f6f740889cba8780d7c4ffe05c
+  private static final String THREAD_GARBAGE_COLLECTOR_OLD = "Service Thread";
+  /**
+   * Returns whether the passed-in thread looks like the garbage collection, both for newer and
+   * older versions of Bazel. See
+   * https://github.com/bazelbuild/bazel/commit/a03674e6297ed5f6f740889cba8780d7c4ffe05c for when
+   * the naming of the garbage collection thread was changed.
+   *
+   * @param thread the thread to check
+   * @return whether the thread looks like it is the garbage collection thread
+   */
+  public static boolean isGarbageCollectorThread(ProfileThread thread) {
```

Comment:
A Constants class feels like the wrong place for a function like this, especially one that takes a thread. Would it be more appropriate in `BazelProfile` itself? It might also be good to check the Bazel version and look for the appropriate name, rather than for either the old or the new.

Reply:
Sadly, the JSON profile does not include the Bazel version as part of the `otherData` field (nor others). Hence, the distinction cannot be precise, rather than having a fallback to older schemas.

In any case, I also find `BazelProfileConstants` inappropriate for a matcher. However, I understand the goal of this class is to put all Bazel-specific keywords together. I wonder if `BazelProfileKeywords` is a better fit as class name.

Reply:
I had these methods in the according classes before (`BazelProfile` and `GarbageCollectionStatsDataProvider`). I decided to move them, though, to prevent people from accidentally using one of these constants without also checking for the second one.

Joseda is also right in that the Bazel profile unfortunately does not include data on which version of Bazel was used when it was generated.

Reply:
I've moved the methods over to `BazelProfile` now and adjusted the code accordingly. In particular, by reducing the visibility of the constants, I think that's sufficient safeguard against them being used.

Reply:
PTAL.